applications/FluidDynamicsApplication/custom_processes/mass_conservation_check_process.cpp

```diff
@@ -0,0 +1,189 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    under supervision of Ruben Zorrilla
+//
+//
+
+// System includes
+#include <fstream>
+
+// External includes
+
+// Project includes
+#include "utilities/openmp_utils.h"
+#include "processes/find_nodal_h_process.h"
+
+// Application includes
+#include "mass_conservation_check_process.h"
+
+
+namespace Kratos
+{
+
+/* Public functions *******************************************************/
+
+MassConservationCheckProcess::MassConservationCheckProcess(
+        ModelPart& rModelPart, 
+        const int MassComputationFreq,
+        const bool CompareToInitial, 
+        const bool WriteToLogFile)
+    : Process(), mrModelPart(rModelPart) {
+
+    mMassComputationFreq = MassComputationFreq;
+    mCompareToInitial = CompareToInitial;
+    mWriteToLogFile = WriteToLogFile;
+}
+
+
+MassConservationCheckProcess::MassConservationCheckProcess(
+    ModelPart& rModelPart,
+    Parameters& rParameters)
+    : Process(), mrModelPart(rModelPart) {
+
+    Parameters default_parameters( R"(
+    {
+        "model_part_name"                        : "default_model_part_name",
+        "mass_computation_frequency"             : 20,
+        "compare_to_initial_values"              : true,
+        "write_to_log_file"                      : true
+    }  )" );
+
+    rParameters.ValidateAndAssignDefaults(default_parameters);
+
+    mMassComputationFreq = rParameters["mass_computation_frequency"].GetInt();
+    mCompareToInitial = rParameters["compare_to_initial_values"].GetBool();
+    mWriteToLogFile = rParameters["write_to_log_file"].GetBool();
+}
+
+
+bool MassConservationCheckProcess::GetUpdateStatus(){
+
+    const int time_step = mrModelPart.GetProcessInfo()[STEP];
+
+    if ( mCompareToInitial && time_step == 1){
+        // getting initial values and storing as reference
+        this->ComputeVolumesOfFluids( mInitialPositiveVolume, mInitialNegativeVolume );
+    }
+
+    if ( time_step % this->mMassComputationFreq == 0){
+        
+        // writing an output at a given frequncy
+        this->ComputeVolumesOfFluids( mCurrentPositiveVolume, mCurrentNegativeVolume );
+
+#ifdef KRATOS_DEBUG
+        std::cout << " --- Volume Checking Process --- " << std::endl;
```

Comment:
leftover cout

Also I think this could be done with an echo-lvl rather than only in debug (it is not crucial for the performance)

Reply:
Whole block of "cout"s has been removed